android/src/main/java/com/instabug/reactlibrary/RNInstabugSessionReplayModule.java

```diff
@@ -97,6 +116,82 @@ public void onSessionReplayLinkReady(@Nullable String link) {
             }
         });
 
+    }
+
+    public WritableArray getNetworkLogsArray(List<SessionMetadata.NetworkLog> networkLogList ){
+        List<SessionMetadata.NetworkLog> networkLogArrayList = networkLogList;
+        
+        WritableArray networkLogs = Arguments.createArray();
+
+        for (SessionMetadata.NetworkLog log : networkLogArrayList) {
+            WritableMap networkLog = Arguments.createMap();
+            networkLog.putString("url", log.getUrl());
+            networkLog.putDouble("duration", log.getDuration());
+            networkLog.putInt("statusCode", log.getStatusCode());
+
+            networkLogs.pushMap(networkLog);
+        }
 
+        return networkLogs;
     }
+
+    private boolean shouldSync = false;
+    private CountDownLatch latch;
+    @ReactMethod
+    public void setSyncCallback() {
+        MainThreadHandler.runOnMainThread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    SessionReplay.setSyncCallback(new SessionSyncListener() {
+                        @Override
+                        public boolean onSessionReadyToSync(@NonNull SessionMetadata sessionMetadata) {
+                            WritableMap params = Arguments.createMap();
+                            params.putString("appVersion",sessionMetadata.getAppVersion());
+                            params.putString("OS",sessionMetadata.getOs());
+                            params.putString("device",sessionMetadata.getDevice());
+                            params.putDouble("sessionDurationInSeconds",(double)sessionMetadata.getSessionDurationInSeconds());
+                            params.putBoolean("hasLinkToAppReview",sessionMetadata.getLinkedToReview());
+                            params.putString("launchType",ArgsRegistry.launchTypeReversed.get(sessionMetadata.getLaunchType()) );
+                            params.putDouble("launchDuration", sessionMetadata.getLaunchDuration());
+                            params.putArray("networkLogs",getNetworkLogsArray(sessionMetadata.getNetworkLogs()));
+                            
+//                              TODO:Add rest of sessionMetadata
+//                            params.putDouble("bugsCount", ??);
+//                            params.putDouble("fatalCrashCount",??);
+//                            params.putDouble("oomCrashCount",??);
+
+                            sendEvent(Constants.IBG_SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION,params);
+
+                            latch = new CountDownLatch(1);
+
+                            try {
+                                latch.await();
+                            } catch (InterruptedException e) {
+                                e.printStackTrace();
+                            }
+
+                            return shouldSync;
```

Comment:
Let's default to returning `true` if an exception is thrown in `latch.await()` or remove the `try-catch` and throw the exception to the native SDK to handle it if possible.

Reply:
@kholood-ea We need to resolve this one before we merge the PR to avoid unintentional return values from the sync callback.